### ISSUE TITLE
Hubspot plugin config (mcrypt error) Mautic 1.2.4

### ISSUE THREAD

**Issue by bhcruz** [mautic/mautic#1391]
Hello guys.
I can't load Hubspot's plugin configuration after 1.2.4 upgrade.
Error 500 ;-(

`[2016-01-28 11:36:18] mautic.WARNING: PHP Warning: mcrypt_decrypt(): Key of size 5 not supported by this algorithm. Only keys of sizes 16, 24 or 32 supported - in file /public/app/bundles/CoreBundle/Helper/EncryptionHelper.php - at line 82 [] []`

PHP 5.6

Any suggestions?

**Comment by dbhurley** [comment]
I believe this warning should not actually kill the plugin. This does seem to be specific to 5.6 as there was an algorithm change to require a certain key length on mcrypt. The key will more than likely need to be regenerated attaching extra bits to meet the acceptable sizes. 

Regardless I don't _believe_ this should be throwing a 500. Can you verify you have no other errors occurring, particularly ones that might relate directly to the HubSpot plugin?

**Comment by bhcruz** [comment]
Hi David.
The 500 error is returned only on Hubspot Plugin Popup config screen.
I can't disable the plugin on admin screen.

But... Mandrill retunring error when I try to send test e-mail and, another info on my log:

```
[2016-01-28 11:41:51] mautic.WARNING: PHP Warning: mcrypt_decrypt(): Key of size 5 not supported by this algorithm. Only keys of sizes 16, 24 or 32 supported - in file /srv/users/serverpilot/apps/mkt/public/app/bundles/CoreBundle/Helper/EncryptionHelper.php - at line 82 [] []
[2016-01-28 11:41:51] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Form\Exception\LogicException: "A form with an empty name cannot have a parent form." at /srv/users/serverpilot/apps/mkt/public/vendor/symfony/form/Symfony/Component/Form/Form.php line 272 {"exception":"[object] (Symfony\\Component\\Form\\Exception\\LogicException(code: 0): A form with an empty name cannot have a parent form. at /srv/users/serverpilot/apps/mkt/public/vendor/symfony/form/Symfony/Component/Form/Form.php:272)"} []
[2016-01-28 12:05:42] mautic.WARNING: PHP Warning: SessionHandler::read(): The session id is too long or contains illegal characters, valid characters are a-z, A-Z, 0-9 and '-,' - in file /srv/users/serverpilot/apps/mkt/public/app/cache/prod/classes.php - at line 411 [] []
[2016-01-28 12:05:42] mautic.WARNING: PHP Warning: SessionHandler::write(): The session id is too long or contains illegal characters, valid characters are a-z, A-Z, 0-9 and '-,' - in file /srv/users/serverpilot/apps/mkt/public/app/cache/prod/classes.php - at line 415 [] []
[2016-01-28 12:48:24] mautic.ERROR: [MAIL ERROR] Mandrill failed to authenticate Log data: !! Mandrill failed to authenticate [] []
```

Can I regenerate the encrypt key wihout corrupt stored data?

**Comment by alanhartless** [comment]
Do you have other plugins configured that are working still? Just verifying if it's hubspot or something more general. We'll have to look at the key warning but I'm with David that I don't thin it's causing the issue.

I think the thing that is breaking it is this: 
```
[2016-01-28 11:41:51] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Form\Exception\LogicException: "A form with an empty name cannot have a parent form." at /srv/users/serverpilot/apps/mkt/public/vendor/symfony/form/Symfony/Component/Form/Form.php line 272 {"exception":"[object] (Symfony\\Component\\Form\\Exception\\LogicException(code: 0): A form with an empty name cannot have a parent form. at /srv/users/serverpilot/apps/mkt/public/vendor/symfony/form/Symfony/Component/Form/Form.php:272)"} []
```
That would indicate that a field is being returned from Hubspot that results that is not handled properly and thus leading to an empty name for a form input field leading to Symfony throwing the exception (uh oh). 

We need to figure out what is being returned by Hubspot that would cause that. 

Can you edit plugins/MauticCrmBundle/Integration/HubspotIntegration.php in your Mautic installation and find line 96. Copy and paste into that blank line:

```
$this->factory->getLogger()->log('error', print_r($leadFields, true));
```

Then click on Hubspot which should result in the error.  Check the logs and report back what's recorded. 

Thanks!
Alan

**Comment by bhcruz** [comment]
Hi Alan.

Other active plugins: Facebook, Google +, LinkedIn and Twitter.
No problems with them.

Only Hubspot with 500 error and Mandrill that do not authenticate.

This is the Mandrill test connection result:
```
Mandrill failed to authenticate Log data: !! Mandrill failed to authenticate
!! Mandrill failed to authenticate

```

And this is the log messages after the file modification:

```
[2016-01-28 17:17:41] mautic.WARNING: PHP Warning: mcrypt_decrypt(): Key of size 5 not supported by this algorithm. Only keys of sizes 16, 24 or 32 supported - in file /srv/users/serverpilot/apps/mkt/public/app/bundles/CoreBundle/Helper/EncryptionHelper.php - at line 82 [] []
[2016-01-28 17:17:41] mautic.ERROR: Array (     [error] => Array         (             [message] => As chaves não estão disponíveis para esta transação ser concluída. Por favor, verifique as configurações e tente novamente.         )  )  [] []
[2016-01-28 17:17:41] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Form\Exception\LogicException: "A form with an empty name cannot have a parent form." at /srv/users/serverpilot/apps/mkt/public/vendor/symfony/form/Symfony/Component/Form/Form.php line 272 {"exception":"[object] (Symfony\\Component\\Form\\Exception\\LogicException(code: 0): A form with an empty name cannot have a parent form. at /srv/users/serverpilot/apps/mkt/public/vendor/symfony/form/Symfony/Component/Form/Form.php:272)"} []
```

I think that can be a problem with my auth keys and mcrypt_decrypt() funcion.

--
Bruno

**Comment by alanhartless** [comment]
Ok, this might be my bad. Looks like a couple lines of code were accidentally committed with a PR that wasn't supposed to be. Digging back through history:

https://github.com/mautic/mautic/blob/master/app/bundles/CoreBundle/Helper/EncryptionHelper.php#L58 and
https://github.com/mautic/mautic/blob/master/app/bundles/CoreBundle/Helper/EncryptionHelper.php#L81

Were being testing to prevent a different warning.  Something about an illegal hex digit or something like that but I don't remember now.  Regardless, I didn't mean for the changes to be committed.

Try reverting back to the previous version to see if that resolves the issue.

https://github.com/mautic/mautic/blob/27501641a1b98db39c52cf653ec0881eb614a3fe/app/bundles/CoreBundle/Helper/EncryptionHelper.php

**Comment by bhcruz** [comment]
Hi Alan,

The reversion to the previous version solved the Hubspot's problem ;)

But not the Mandrill authentication.
```
Mandrill failed to authenticate Log data: !! Mandrill failed to authenticate
!! Mandrill failed to authenticate
```

See the log messages after update:
`[2016-02-01 11:11:44] mautic.WARNING: PHP Warning: SessionHandler::write(): The session id is too long or contains illegal characters, valid characters are a-z, A-Z, 0-9 and '-,' - in file /srv/users/serverpilot/apps/mkt/public/app/cache/prod/classes.php - at line 415 [] []`

**Comment by alanhartless** [comment]
Is that message consistent? Does it happen anywhere in Mautic or just with authenticating with Mandrill?

**Comment by escopecz** [comment]
Someone reported to me that he's getting 
```
mautic.WARNING: PHP Warning: mcrypt_decrypt(): Key of size 5 not supported by this algorithm. Only keys of sizes 16, 24 or 32 supported - in file app/bundles/CoreBundle/Helper/EncryptionHelper.php - at line 82
```
and that the social plugins doesn't work because of it. I can ask for more details.

**Comment by alanhartless** [comment]
John, the PR above should fix that. (#1396)

**Comment by escopecz** [comment]
ah, I overlooked it. Thanks

**Comment by bhcruz** [comment]
Just with authentication with Mandrill.

**Comment by dongilbert** [comment]
This issue has been fixed and closed via #1396
